### PR TITLE
Implement publish confirm mode

### PR DESCRIPTION
**Pull request by fuczak**

### CODE REVIEW

File: src/Rabbitmq/Manager/Manager.php [hunk 1]
```diff
@@ -21,56 +21,61 @@ class Manager {
...
      * @param \Maleficarum\Rabbitmq\Connection\Connection $connection
      * @param string $identifier
      * @param int $mode
      * @param int $priority
+     * @param int $publishConfirmTimeout
      * @throws \InvalidArgumentException
      * @return \Maleficarum\Rabbitmq\Manager\Manager
      */
-    public function addConnection(\Maleficarum\Rabbitmq\Connection\Connection $connection, string $identifier, int $mode, int $priority = 0) : \Maleficarum\Rabbitmq\Manager\Manager {
+    public function addConnection(\Maleficarum\Rabbitmq\Connection\Connection $connection, string $identifier, int $mode, int $priority = 0, int $publishConfirmTimeout = 0) : \Maleficarum\Rabbitmq\Manager\Manager {
```
Comment (bartekmielnik) [inline src/Rabbitmq/Manager/Manager.php:52]: It always seems good to indicate the units in the name of the param, here - I don't know whether it's in seconds or milliseconds or anything else.

Comment (pharaun13) [reply src/Rabbitmq/Manager/Manager.php:52]: This is a fair point.

File: src/Rabbitmq/Manager/Manager.php [hunk 2]
```diff
@@ -109,32 +114,43 @@ public function addCommand(\Maleficarum\Command\AbstractCommand $command, string
...
             [$command->toJSON(), ['delivery_mode' => 2, 'application_headers' => $applicationHeaders]]);
         $channel = $connection->getChannel();
+
+        // establish publish confirm mode if necessary
+        $publishConfirmTimeout > 0 and $channel->confirm_select();
+
         $channel->basic_publish($message, $connection->getExchangeName(), $connection->getQueueName());
+
+        // wait for publish confirm
+        $channel->wait_for_pending_acks($publishConfirmTimeout);
```
Comment (pharaun13) [inline src/Rabbitmq/Manager/Manager.php]: If the confirm_select() is conditional why isn't wait_for_pending_acks() conditional as well?

File: src/Rabbitmq/Manager/Manager.php [hunk 3]
```diff
@@ -167,12 +186,20 @@ public function addCommands(array $commands, string $connectionIdentifier, array
...
         $applicationHeaders = \Maleficarum\Ioc\Container::get('PhpAmqpLib\Wire\AMQPTable', [$commandsHeaders]);
         foreach ($commands as $command) {
             $message = \Maleficarum\Ioc\Container::get('PhpAmqpLib\Message\AMQPMessage', [$command->toJSON(), ['delivery_mode' => 2, 'application_headers' => $applicationHeaders]]);
             $channel->batch_basic_publish($message, $connection->getExchangeName(), $connection->getQueueName());
         }
         $channel->publish_batch();
+
+        // wait for publish confirm
+        $channel->wait_for_pending_acks($publishConfirmTimeout);
+
```
Comment (pharaun13) [inline src/Rabbitmq/Manager/Manager.php]: If the confirm_select() is conditional why isn't wait_for_pending_acks() conditional as well?

**Pull request merged**